tests/integration/protected_product.t

```diff
@@ -0,0 +1,228 @@
+			serving_size => '20 g',
+			ingredients_text_fr => "Farine de blé, eau, sel, sucre",
+			labels => "Bio, Max Havelaar",
+			nutriment_salt => '0.7',
+			nutriment_salt_unit => 'mg',
+			nutriment_fat => '12.7',
+			nutriment_sugars => '7',
+		}
+	},
+	{
+		test_case => 'get-edited-unprotected-product-api-v2',
+		method => 'GET',
+		path => '/api/v2/product/0200000000034',
+	},
+	{
+		test_case => 'edit-protected-product-api-v2',
+		method => 'POST',
+		path => '/cgi/product_jqm_multilingual.pl',    # API v0/v2
+		form => {
+			user_id => "tests",
+			password => "testtest",
+			cc => "be",
+			lc => "fr",
+			code => "0200000000035",
+			product_name_en => "Changed product name",
+			product_name_fr => "New French product name",
+			categories => "Changed category",
+			quantity => "250 g",
+			serving_size => '20 g',
+			ingredients_text_fr => "Farine de blé, eau, sel, sucre",
+			labels => "Bio, Max Havelaar",
+			nutriment_salt => '0.7',
+			nutriment_salt_unit => 'mg',
+			nutriment_fat => '12.7',
+			nutriment_sugars => '7',
+		}
+	},
+	{
+		test_case => 'get-edited-protected-product-api-v2',
+		method => 'GET',
+		path => '/api/v2/product/0200000000035',
+	},
+	# Test with the /cgi/product.pl web form
+	{
+		test_case => 'edit-unprotected-product-web-form',
+		method => 'POST',
+		path => '/cgi/product.pl',
+		form => {
+			type => "edit",
+			action => "process",
+			sorted_langs => "en,fr",
+			user_id => "tests",
+			password => "testtest",
+			cc => "be",
+			lc => "fr",
+			code => "0200000000134",
+			product_name_en => "Changed product name",
+			product_name_fr => "New French product name",
+			categories => "Changed category",
+			quantity => "250 g",
+			serving_size => '20 g',
+			ingredients_text_fr => "Farine de blé, eau, sel, sucre",
+			labels => "Bio, Max Havelaar",
+			nutriment_salt => '0.7',
+			nutriment_salt_unit => 'mg',
+			nutriment_fat => '12.7',
+			nutriment_sugars => '7',
+		},
+		expected_type => 'html',
+	},
+	{
+		test_case => 'get-edited-unprotected-product-web-form',
+		method => 'GET',
+		path => '/api/v2/product/0200000000134',
+	},
+	{
+		test_case => 'edit-protected-product-web-form',
+		method => 'POST',
+		path => '/cgi/product.pl',
+		form => {
+			type => "edit",
+			action => "process",
+			sorted_langs => "en,fr",
+			user_id => "tests",
+			password => "testtest",
+			cc => "be",
+			lc => "fr",
+			code => "0200000000135",
+			product_name_en => "Changed product name",
+			product_name_fr => "New French product name",
+			categories => "Changed category",
+			quantity => "250 g",
+			serving_size => '20 g',
+			ingredients_text_fr => "Farine de blé, eau, sel, sucre",
+			labels => "Bio, Max Havelaar",
+			nutriment_salt => '0.7',
+			nutriment_salt_unit => 'mg',
+			nutriment_fat => '12.7',
+			nutriment_sugars => '7',
+		},
+		expected_type => 'html',
+	},
+	{
+		test_case => 'get-edited-protected-product-web-form',
+		method => 'GET',
+		path => '/api/v2/product/0200000000135',
+	},
+];
```

Comment:
You did not test,if it works for moderators on protected products.

Reply:
@alexgarel I added some tests for moderators.